zstacklib/zstacklib/utils/plugin.py

```
@@ -67,6 +67,8 @@ class TaskDaemon(object):
     '''
 
     def __init__(self, task_spec, task_name, timeout=0):
+        if self.__class__.__name__ != TaskDaemon.__name__ and self.__class__.__dict__.has_key("__exit__"):
+            raise Exception("method __exit__ cannot be overridden")
```

Ruff (0.14.8)

Create your own exception

(TRY002)